### PR TITLE
Ensure consistent errors

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This PR updates the codebase to use `pkg/errors.Errorf` in place of `fmt.Errorf`
the format strings have been updated to use `%s` (the intended mechanism for handling errors in format strings)

### CODE REVIEW

File: test/e2e/mongodbtests/mongodbtests.go [hunk 1]
```diff
@@ -200,7 +202,7 @@ func Connectivity(mdb *mdbv1.MongoDB, username, password string) func(t *testing
 			Username:      username,
 			Password:      password,
 		})); err != nil {
-			t.Fatal(fmt.Sprintf("Error connecting to MongoDB deployment: %+v", err))
+			t.Fatal(fmt.Sprintf("Error connecting to MongoDB deployment: %s", err))
```
Comment (bznein) [inline test/e2e/mongodbtests/mongodbtests.go:205]: [q] is there a specific reason here we use `fmt.Sprintf`? A few lines below we have the same structure (`t.Fatal`) with inside a `errors.Errorf`

File: test/e2e/mongodbtests/mongodbtests.go [hunk 2]
```diff
@@ -288,7 +290,7 @@ func IsReachableDuringWithConnection(mdb *mdbv1.MongoDB, interval time.Duration,
 					return
 				case <-time.After(interval):
 					if err := connectFunc(); err != nil {
-						t.Fatal(fmt.Sprintf("error reaching MongoDB deployment: %+v", err))
+						t.Fatal(fmt.Sprintf("error reaching MongoDB deployment: %s", err))
```
Comment (bznein) [inline test/e2e/mongodbtests/mongodbtests.go:293]: Same as the previous comment

File: test/e2e/replica_set_tls/replica_set_tls_test.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ func TestReplicaSetTLS(t *testing.T) {
 	}
 
 	if err := setup.CreateTLSResources(mdb.Namespace, ctx); err != nil {
-		t.Fatalf("Failed to set up TLS resources: %+v", err)
+		t.Fatalf("Failed to set up TLS resources: %s", err)
```
Comment (bznein) [inline test/e2e/replica_set_tls/replica_set_tls_test.go:33]: Seeing this and referring to my other two comments, maybe using `t.Fatalf` every time would be better? I feel we have a mixture of three different approaches at the moment:

`t.Fatalf`
`t.Fatal(fmt.Sprintf(...))`
`t.Fatal(errors.Errorf(...))`

**Review by bznein — APPROVED** [review]
LGTM but unless I misunderstood something we can achieve even more consistency in our error handling

File: test/e2e/replica_set_tls/replica_set_tls_test.go (hunk 3 above)
Comment (chatton) [reply test/e2e/replica_set_tls/replica_set_tls_test.go:33]: yes I agree! I will update to use one. I think `t.Fatalf` is the most straightfoward, what do you think?

Comment (bznein) [reply test/e2e/replica_set_tls/replica_set_tls_test.go:33]: I think so! No need to nest function calls!

**Pull request merged**